### PR TITLE
Fix typos in greenspline doc and add equations

### PR DESCRIPTION
**Pull request by PaulWessel**
More use of math is not simple in rst.

### CODE REVIEW

File: doc/rst/source/greenspline.rst [hunk 1]
```diff
@@ -44,24 +44,45 @@ Synopsis
...
+.. math::
+
+    w(\mathbf{x}) = T(\mathbf{x}) + \sum_{j=1}^{n} \alpha_j g(\mathbf{x}; \mathbf{x}'),
+
+where :math:`\mathbf{x}` is the output location, *n* is the number of points,
+:math:`T(\mathbf{x})` is a trend function, and :math:`\alpha_j` are the *n*
+unknown weights we must solve for. Typically, :math:`T(\mathbf{x})` is a linear
+or planar trend (Cartesian geometries) or mean value (spherical surface) and a
+least-squares solution is determined and removed from the data, yielding data
+residuals (:math:`\Delta w_i = w_i - T(\mathbf{x}_i)`; these are then
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:63]:
```suggestion
residuals (:math:`\Delta w_i = w_i - T(\mathbf{x}_i)`); these are then
```

File: doc/rst/source/greenspline.rst [hunk 2]
```diff
@@ -398,12 +420,12 @@ Considerations
...
    place restrictions on how large data sets you can process with
-   **greenspline**. Pre-processing your data with doc:`blockmean`,
-   doc:`blockmedian`, or doc:`blockmode` is recommended to avoid aliasing and
+   **greenspline**. Pre-processing your data with :doc:`blockmean`,
+   :doc:`blockmedian`, or :doc:`blockmode` is recommended to avoid aliasing and
    may also control the size of *n*. For information, if *n* = 1024 then
    only 8 Mb memory is needed, but for *n* = 10240 we need 800 Mb. Note
    that **greenspline** is fully 64-bit compliant if compiled as such.
-   For spherical data you may consider decimating using doc:`gmtspatial`
+   For spherical data you may consider decimating using ;doc:`gmtspatial`
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:428]:
```suggestion
   For spherical data you may consider decimating using :doc:`gmtspatial`
```

File: doc/rst/source/greenspline.rst [hunk 3]
```diff
@@ -44,24 +44,45 @@ Synopsis
...
-solving the *n* by *n* linear system *w*\ (*j*) = sum-over-*i*
-{*c*\ (*i*) G(\ **x**\ (*j*); **x**\ (*i*))}, for *j* = 1, *n*; this
+Cartesian coordinates or spherical surface coordinates. Mathematically, the solution is composed as
+
+
+.. math::
+
+    w(\mathbf{x}) = T(\mathbf{x}) + \sum_{j=1}^{n} \alpha_j g(\mathbf{x}; \mathbf{x}'),
+
+where :math:`\mathbf{x}` is the output location, *n* is the number of points,
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:58]:
```suggestion
where :math:`\mathbf{x}` is the output location, :math:`n` is the number of points,
```

File: doc/rst/source/greenspline.rst [hunk 4]
```diff
@@ -44,24 +44,45 @@ Synopsis
...
+yielding a :math:`n \times n` linear system to be solved for the coefficients.
+Finally, away from the data constraints the Green's function must satisfy
+
+.. math::
+
+    \nabla^2  \left [ \nabla^2 - p^2 \right ] g(\mathbf{x}; \mathbf{x}') = \delta (\mathbf{x} - \mathbf{x}'),
+
+
+where :math:`\nabla^2` is the Laplacian operator, :math:`\delta` is the
+Dirac Delta function, and *p* is the tension (if desired). This
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:81]:
```suggestion
Dirac Delta function, and :math:`p` is the tension (if desired). This
```

File: doc/rst/source/greenspline.rst [hunk 5]
```diff
@@ -78,11 +99,12 @@ Optional Arguments
 .. _-A:
 
 **-A**\ *gradfile*\ **+f**\ **1**\|\ **2**\|\ **3**\|\ **4**\|\ **5**
-    The solution will partly be constrained by surface gradients **v** =
-    *v*\ \*\ **n**, where *v* is the gradient magnitude and **n** its
-    unit vector direction. The gradient direction may be specified
-    either by Cartesian components (either unit vector **n** and
-    magnitude *v* separately or gradient components **v** directly) or
+    The solution will partly be constrained by surface gradients
+    :math:`\mathbf{v} = v \hat{\mathbf{n}}`, where *v* is the gradient
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:103]:
```suggestion
    :math:`\mathbf{v} = v \hat{\mathbf{n}}`, where :math:`v` is the gradient
```

File: doc/rst/source/greenspline.rst [hunk 6]
```diff
@@ -78,11 +99,12 @@ Optional Arguments
...
+    magnitude and :math:`\hat{\mathbf{n}}` its unit vector direction.
+    The gradient direction may be specified either by Cartesian components
+    (either unit vector :math:`\hat{\mathbf{n}}` and magnitude *v* separately
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:106]:
```suggestion
    (either unit vector :math:`\hat{\mathbf{n}}` and magnitude :math:`v` separately
```

File: doc/rst/source/greenspline.rst [hunk 7]
```diff
@@ -95,7 +117,7 @@ Optional Arguments
     or 3-D data: **3**: records contain **x**, *direction(s)*, *v*
     (*direction(s)* in degrees are measured counter-clockwise from the
     horizontal (and for 3-D the vertical axis). **4**: records contain
-    **x**, **v**. **5**: records contain **x**, **n**, *v*.
+    **x**, **v**. **5**: records contain **x**, :math:`\hat{\mathbf{n}}`, *v*.
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:120]:
```suggestion
    **x**, **v**. **5**: records contain **x**, :math:`\hat{\mathbf{n}}`, :math:`v`.
```

File: doc/rst/source/greenspline.rst [hunk 8]
```diff
@@ -95,7 +117,7 @@ Optional Arguments
     or 3-D data: **3**: records contain **x**, *direction(s)*, *v*
     (*direction(s)* in degrees are measured counter-clockwise from the
     horizontal (and for 3-D the vertical axis). **4**: records contain
```
Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:119]: Is the *gradient* in options 0,1, and 2 the same as the gradient magnitude *v*? If so, I think all should use the same notation.

Comment (meghanrjones) [inline doc/rst/source/greenspline.rst:119]:
```suggestion
    (*direction(s)* in degrees are measured counter-clockwise from the
    horizontal (and for 3-D the vertical axis)). **4**: records contain
```

**Review by meghanrjones — APPROVED** [review]

File: doc/rst/source/greenspline.rst (hunk 8 above)
Comment (PaulWessel) [reply doc/rst/source/greenspline.rst]: Fixed

**Pull request merged**